packages/@aws-cdk/aws-ec2/lib/launch-template.ts

```diff
@@ -0,0 +1,688 @@
+   * reference the DEFAULT version of the template.
+   */
+  public static readonly DEFAULT_VERSION: string = '$Default';
+}
+
+/**
+ * Attributes for an imported LaunchTemplate.
+ */
+export interface LaunchTemplateAttributes {
+  /**
+   * The version number of this launch template to use
+   *
+   * @default Version: "$Default"
+   */
+  readonly versionNumber?: string;
+
+  /**
+   * The identifier of the Launch Template
+   *
+   * Exactly one of `launchTemplateId` and `launchTemplateName` may be set.
+   *
+   * @default None
+   */
+  readonly launchTemplateId?: string;
+
+  /**
+   * The name of the Launch Template
+   *
+   * Exactly one of `launchTemplateId` and `launchTemplateName` may be set.
+   *
+   * @default None
+   */
+  readonly launchTemplateName?: string;
+}
+
+/**
+ * This represents an EC2 LaunchTemplate.
+ *
+ * @see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-launch-templates.html
+ */
+export class LaunchTemplate extends Resource implements ILaunchTemplate, iam.IGrantable, IConnectable {
+  /**
+   * Import an existing LaunchTemplate.
+   */
+  public static fromLaunchTemplateAttributes(scope: Construct, id: string, attrs: LaunchTemplateAttributes): ILaunchTemplate {
+    const haveId = Boolean(attrs.launchTemplateId);
+    const haveName = Boolean(attrs.launchTemplateName);
+    if (haveId == haveName) {
+      throw new Error('LaunchTemplate.fromLaunchTemplateAttributes() requires exactly one of launchTemplateId or launchTemplateName be provided.');
+    }
+
+    class Import extends Resource implements ILaunchTemplate {
+      public readonly versionNumber = attrs.versionNumber ?? LaunchTemplateSpecialVersions.DEFAULT_VERSION;
+      public readonly launchTemplateId? = attrs.launchTemplateId;
+      public readonly launchTemplateName? = attrs.launchTemplateName;
+    }
+    return new Import(scope, id);
+  }
+
+  // ============================================
+  //   Members for ILaunchTemplate interface
+  /**
+   * @inheritdoc
+   */
+  public readonly versionNumber: string;
+
+  /**
+   * @inheritdoc
+   */
+  public readonly launchTemplateId?: string;
+
+  /**
+   * @inheritdoc
+   */
+  public readonly launchTemplateName?: string;
+
+  // =============================================
+  //   Data members
+
+  /**
+   * The default version for the launch template.
+   *
+   * @attribute
+   */
+  public readonly defaultVersionNumber: string;
+
+  /**
+   * The latest version of the launch template.
+   *
+   * @attribute
+   */
+  public readonly latestVersionNumber: string;
+
+  /**
+   * IAM Role assumed by instances that are launched from this template.
+   */
+  public readonly role: iam.IRole;
+
+  /**
+   * Principal to grant permissions to.
+   */
+  public readonly grantPrincipal: iam.IPrincipal;
+
+  /**
+   * Allows specifying security group connections for the instance.
+   *
+   * Note: If the LaunchTemplate was not created with a SecurityGroup, then this connections
```

Comment:
I think I would support having it throw an exception, and same for `role` and `grantPrincipal`:

```ts
public get connections(): Connections {
  if (!this._connections) {
    throw new Error('oh no');
  }
  return this._connections;
}
```